insights/core/__init__.py

```diff
@@ -284,6 +285,36 @@ def get(self, item, default=None):
         return self.data.get(item, default)
 
 
+def get_xmlns(self, root):
+        return root.tag.strip("{").split("}")[0] if all(c in root.tag for c in ["{", "}"]) else ""
+
+
+class XMLParser(Parser):
+    """
+    A parser class that reads XML files.  Base your own parser on this.
+    """
+
+    def parse_content(self, content):
+        root = ET.fromstring('\n'.join(content))
+        self.dom = root
+        self.xmlns = self.get_xmlns(root)
+
+    def get_xmlns(self, root):
+        return root.tag.strip("{").split("}")[0] if all(c in root.tag for c in ["{", "}"]) else ""
+
+    def get_elements(self, element):
```

Comment:
How about adding the default `namespaces` as the default argument:
`def get_elements(self, element, xmlns=self.xmlns)`?